### PR TITLE
update collection-level physloc notes to new codes

### PR DESCRIPTION
**Pull request by regineheberlein**

### CODE REVIEW

File: helper_methods.rb [hunk 1]
```diff
@@ -338,19 +337,55 @@ def add_revision_statement(uri, text)
 end
 
 #add a maintenance statement to agent records
-def add_maintenance_history(uri, text)
-  record = @client.get(uri).parsed
+def add_maintenance_history(record, text)
+  #record = @client.get(uri).parsed
```
Comment (rladdusaw) [inline helper_methods.rb:341]: Is this being kept so it can be reverted in the future? Maybe a comment would be helpful?

File: helper_methods.rb [hunk 2]
```diff
@@ -338,19 +337,55 @@ def add_revision_statement(uri, text)
...
     "descriptive_note"=>text.to_s,
-    "created_by"=>"system",
+    "created_by"=>"aspace_helpers",
     "publish"=>true,
     "jsonmodel_type"=>"agent_maintenance_history"
   }
 
-  post = @client.post(uri, record.to_json)
-  puts post.body
+#
```
Comment (rladdusaw) [inline helper_methods.rb:353]: Can the # be removed?

File: helper_methods.rb [hunk 3]
```diff
@@ -338,19 +337,55 @@ def add_revision_statement(uri, text)
...
+      query: {
+       all_ids: true
+      }}).parsed
+    ids_by_endpoint = ids_by_endpoint.flatten!
+    ids_by_endpoint.each do |id|
+      @uris << "/#{endpoint}/#{id}"
+    end
+  end #close resources_endpoints.each
+  @uris
+end #close method
```
Comment (rladdusaw) [inline helper_methods.rb:391]: These comments don't add much to readability and can be removed.

File: helper_methods.rb [hunk 4]
```diff
@@ -338,19 +337,55 @@ def add_revision_statement(uri, text)
...
+  rrecord['revision_statements'] << {
+    "date"=>"#{Time.now}",
+    "created_by"=>"system",
+    "last_modified_by"=>"aspace_helpers",
+    "create_time"=>"#{Time.now}",
+    "description"=>text.to_s,
+    "publish"=>true,
+    "jsonmodel_type"=>"revision_statement"
+  }
+  end
```
Comment (rladdusaw) [inline helper_methods.rb:367]: The end's indentation doesn't match the start of the method.

**Review by rladdusaw — COMMENTED** [review]
Code looks good, just some formatting suggestions.

**Comment by regineheberlein** [comment]
closes #255 and #259

File: reports/physlocs_used_by_repos/physlocs_used_by_repos.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  records.map do |record|
+    record['response']['docs'].select do |container|
+    top_container_location_code = container['location_display_string_u_sstr'].nil? ? '' : container['location_display_string_u_sstr'][0].gsub(/(^.+\[)(.+)(\].*)/, '\2')
+    top_container_location_codes << {'location' => top_container_location_code, 'repo' => container['repository'].gsub(/\/repositories\//, '') }
+  end
+end
+#tally location codes
+group_by_location = top_container_location_codes.group_by { |hash| hash['location'] }.map do |physloc, top_container_location_codes|
+  repo = top_container_location_codes.map { |hash| hash['repo']}
+  {'physloc' => physloc, 'repo' => repo.tally}
```
Comment (sandbergja) [inline reports/physlocs_used_by_repos/physlocs_used_by_repos.rb:31]: Cool!

**Review by sandbergja — APPROVED** [review]
Thanks, @regineheberlein

File: reports/physlocs_used_by_repos/physlocs_used_by_repos.rb (hunk 5 above)
Comment (regineheberlein) [reply reports/physlocs_used_by_repos/physlocs_used_by_repos.rb:31]: right? :-)

**Pull request merged**